### PR TITLE
[ios][background-fetch] Fixed unnecessary `location` capability in UIBackgroundModes leading to App Store rejection for apps not using location

### PR DESCRIPTION
**Pull request by martinstoyanov-brex**
# Why

With version 9.1.0+ expo-background-fetch and expo-task-manager add the `location` capability to UIBackgroundModes on iOS upon expo-eject. If an app uses these packages but does not require persistent location it can lead to App Store rejection for apps including the capability without user-facing feature requiring location.

# How

I removed the addition of 'location' by the config plugins as it is unnecessary to enable background fetch and at worst can lead to App Store rejection if not caught. This updates the packages to version 9.2.1

# Test Plan

Tested manually after running `expo eject`, by opening the `ios/app/Info.plist` and verifying that `location` is not added anymore under `UIBackgroundModes`.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by byCedric** [comment]
cc @EvanBacon, if we do this, we might be able to add it back with the `expo-location` config plugin using a prop like `useBackgroundLocation: true`?

File: packages/expo-task-manager/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "expo-task-manager",
-  "version": "9.2.0",
+  "version": "9.2.1",
```
Comment (brentvatne) [inline packages/expo-task-manager/package.json:3]:
```suggestion
  "version": "9.2.0",
```

Comment (brentvatne) [reply packages/expo-task-manager/package.json:3]: we'll update the package version automatically when we publish it :)

File: packages/expo-background-fetch/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "expo-background-fetch",
-  "version": "9.2.0",
+  "version": "9.2.1",
```
Comment (brentvatne) [inline packages/expo-background-fetch/package.json:3]:
```suggestion
  "version": "9.2.0",
```

File: packages/expo-background-fetch/CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +18,11 @@
 
 - Migrated from `@unimodules/core` to `expo-modules-core`. ([#13757](https://github.com/expo/expo/pull/13757) by [@tsapeta](https://github.com/tsapeta))
 
+## 9.2.1 — 2021-08-02
```
Comment (brentvatne) [inline packages/expo-background-fetch/CHANGELOG.md:21]: can you move this under bug fixes for unpublished? we handle moving the entries under the correct version when we publish

**Review by brentvatne — APPROVED** [review]

File: packages/expo-background-fetch/CHANGELOG.md (hunk 3 above)
Comment (martinstoyanov-brex) [reply packages/expo-background-fetch/CHANGELOG.md]: Changes made. :)

Thank you!

**Comment by EvanBacon** [comment]
Fixed in https://github.com/expo/expo/issues/14142

**Pull request closed without merging**